x-pack/plugin/esql/compute/gen/src/main/java/org/elasticsearch/compute/gen/EvaluatorImplementer.java

```diff
@@ -68,12 +70,12 @@ public EvaluatorImplementer(
         List<TypeMirror> warnExceptions
     ) {
         this.declarationType = (TypeElement) processFunction.getEnclosingElement();
-        this.processFunction = new ProcessFunction(elements, types, processFunction, warnExceptions);
-
         this.implementation = ClassName.get(
             elements.getPackageOf(declarationType).toString(),
             declarationType.getSimpleName() + extraName + "Evaluator"
         );
+        this.isIn = declarationType.getSimpleName().toString().equalsIgnoreCase("In");
```

Comment:
There might be a better way to detect whether this is an In predicate?

Reply:
Could the `blockStyle` invocation work? That's less generated code, but might be ok?

Reply:
> Could the `blockStyle` invocation work? That's less generated code, but might be ok?

It looks like ArrayProcessFunctionArg does quite work with block type inputs, I'll see if I can make it work and if it is more or less complicated than the current approach.

Reply:
What about `BlockProcessFunctionArg`? That's kind of "I'll do it manually" for process.

Reply:
> What about `BlockProcessFunctionArg`? That's kind of "I'll do it manually" for process.

BlockProcessFunctionArg only works block type, it doesn't support block type array... the inlist is an array.